docs/deployment/pull-backup.rst

```diff
@@ -0,0 +1,114 @@
+.. include:: ../global.rst.inc
+.. highlight:: none
+
+Backing up in pull mode
+=======================
+
+Assuming you have a pull backup system set up with borg, where a backup server
+pulls the data from the target via sshfs.
+
+In this mode, the backup client's file system is mounted remotely on the backup
+server. Pull mode is even possible if the SSH connection must be established by
+the client via a remote tunnel. Other network file systems like NFS or SMB could
+be used as well, but sshfs is very simple to set up and probably the most secure
+one.
+
+There are some restrictions caused by sshfs. For example, unless you define UID
+and GID mappings when mounting via sshfs, owners and groups of the mounted file
+system will probably change, and you may not have access to those files if
+BorgBackup is not run with root privileges.
+
+The sshfs is a FUSE file system and uses the SFTP protocol, so there may be also
+other unsupported features that the actual implementations of sshfs, libfuse and
+sftp on the backup server do not support, like file name encodings, ACLs, xattrs
+or bsdflags. So there is no guarantee that you are able to restore a system
+completely in every aspect from such a backup.
+
+To mount the client's root file system you will need root access to the client.
+This contradicts to the usual threat model of BorgBackup, where clients don't
+need to trust the backup server can encrypt their data. In pull mode the server
+(when logged in as root) could cause unlimited damage to the client. Therefore,
+pull mode should be used only from servers you own or trust yourself!
+
+Creating a backup
+-----------------
+
+In this approach the client file system is simply mounted and then backed up.
+Note that the backup is created from within the mount point, so that all files
+in the archive have their original paths (otherwise they would be backed up with
+the mount point prefix, e.g. /mnt/sshfs/bin/bash instead of /bin/bash).
+
+::
+
+    # Mount client root file system.
+    mkdir /mnt/sshfs
+    sshfs root@host:/ /mnt/sshfs
+    # Change into mount dir and back up.
+    cd /mnt/sshfs
+    borg create /borg/to/repo::archive
+    # Unmount client.
+    cd ~
+    fusermount -u /mnt/sshfs
+
+Restoring a full backup
+-----------------------
+
+In case of a restore there is no direct way for borg to know the correct UID:GID
+of files, because it uses the ``/etc/passwd`` and ``/etc/group`` of the backup
+server. Using the archived UIDs/GIDs might lead to wrong results, because after
```

Comment:
Let's assume I have backed up ``/home/fantasya``, UID is 1000. After a crash I reinstalled, but with UID 1001. So restoring would set my homedir with UID 1000 - bad. This will be the case with --numeric-owner. But without --numeric-owner? I read that Borg also saves owner name. Would ``extract`` try to set owner to ``fantasya`` if it is available?

Reply:
This question could be a new FAQ: "Does Borg restore owner/group by ID or by name?"
I assume Borg first checks if a name exists in /etc/passwd or /etc/group, then sets "chown <name> <file>" or so. If not found, then sets "chown <uid> <file>". Is that correct?

Reply:
my comment was based on the idea of a full system backup and a full restore.
in that case, everything will be consistent if one just restores the numeric uids/gids as present in the archive.

Reply:
even in the other case (not doing a full backup/restore), I guess, at backup time, one wants to avoid looking up user/group names on the backup server for the uids/gids seen on the client files - there's a quite big chance of mismatches.

at restore time, it depends.
- full backup / full restore will be automatically consistent when using only the numeric ids.
- if the original /etc/passwd and groups is lost and a new different one is effective now, one of course needs to use name-based mapping and all user/group names need to be at least present.

another problem is you can't mix these. if you did not backup the names, you can't use them at restore time.

i'm a fan of full backups. :-)

Reply:
I think you are right, but I'm a bit confused now. When we do a full restore with --numeric-owner, then why do we have to create a chroot? It should be sufficient to mount to /mnt/sshfs, then cd /mnt/sshfs, then borg extract. Am I missing something?

Reply:
Thank you. I have tried to add this explanations and made a separation between partial and full restores. Also added a note for the ``export-tar`` variant (loss of features).